mantidproject/mantid
# Architecture for Muon Analysis version 2

Muon Analysis version 2 will be a fully MVP code. This PR is for the basic architecture of the GUI, this includes a load widget at the top, some tabs in the centre and a help widget at the bottom. To demonstrate the GUI we are only using dummy widgets (button or a label). 

This does not include any tests as the only part of the code we will keep is the dock_widget and dock_view, which just wrap a pyqt widget (so we don't need to check it works).

**To test:**


To have muon analysis 2 as an interface you will need add the following to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L23    `Muon/Muon_Analysis_2.py`

Then build mantid

Once it has built open mantid and open Muon Analysis 2
A GUI will appear, it will have labels at the top and bottom, then some tabs in the centre. 
If you click the square in the top right corner of the tab it should be ejected.
Eject 2 of the 3 tabs and position them so you can see them all
Press one of the buttons and the label on tab two will change (depending on which button you pressed the label will be different)
Make sure you can put the tabs back into the GUI (pressing the square in the corner should do the trick)

Eject all of the tabs
Close the GUI main window
All of the tabs should also close

Fixes #20941  

**Release Notes** 


---
Will not be available to the user, so not in the release notes
#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
